packages/lesswrong/components/hooks/useUnreadNotifications.ts

```diff
@@ -55,38 +86,73 @@ export function useUnreadNotifications(): {
   
   const refetchBoth = useCallback(async () => {
     if (currentUser?._id) {
-      await Promise.all([
-        refetchCounts(),
-        refetchNotifications(),
-      ]);
+      void refetchNotifications();
+
+      const newCounts = await refetchCounts();
+      updateFavicon(newCounts.data);
     }
   }, [currentUser?._id, refetchCounts, refetchNotifications]);
 
+  // Subscribe to localStorage change events. The localStorage key
+  // "notificationsCheckedAt" contains a date; when the user checks
+  // notifications they write that key. If there's a badge notification,
+  // other tabs handle the event by clearing the badge.
+  useEffect(() => {
+    const storageEventListener = (event: StorageEvent) => {
+      if (
+        event.key === notificationsCheckedAtLocalStorageKey
+        && event.newValue
+      ) {
+        const newCheckedAt = new Date(event.newValue);
+        if (checkedAt && newCheckedAt>checkedAt) {
+          void refetchCounts();
```

Comment:
I don't understand why this one calls refetchCounts and doesn't call updateFavicon directly, but above in refetchBoth, we use the result of refetchCounts to update the favicon explicitly.

Reply:
If this clears the favicon then to my read that's because refetchCounts causes the onCompleted callback to fire. If that's the case then I don't think you need the updateFavicon call in refetchBoth.

Reply:
Just to check: you have tested that this does clear the favicon of other tabs?

Reply:
Yes, I've tested the clearing; each tab winds up doing this refetch and updating its own favicon separately.